editor/editor_node.cpp

```diff
@@ -1042,34 +1026,54 @@ void EditorNode::_resources_reimported(const Vector<String> &p_resources) {
 }
 
 void EditorNode::_sources_changed(bool p_exist) {
-	if (waiting_for_first_scan) {
-		waiting_for_first_scan = false;
+	if (!waiting_for_first_scan) {
+		return;
+	}
+
+	waiting_for_first_scan = false;
+
+	OS::get_singleton()->benchmark_end_measure("Editor", "First Scan");
+
+	// Reload the global shader variables, but this time
+	// loading textures, as they are now properly imported.
+	RenderingServer::get_singleton()->global_shader_parameters_load_settings(true);
+
+	_load_editor_layout();
 
-		OS::get_singleton()->benchmark_end_measure("Editor", "First Scan");
+	if (!defer_load_scene.is_empty()) {
+		OS::get_singleton()->benchmark_begin_measure("Editor", "Load Scene");
 
-		// Reload the global shader variables, but this time
-		// loading textures, as they are now properly imported.
-		RenderingServer::get_singleton()->global_shader_parameters_load_settings(true);
+		load_scene(defer_load_scene);
+		defer_load_scene = "";
 
-		_load_editor_layout();
+		OS::get_singleton()->benchmark_end_measure("Editor", "Load Scene");
+		OS::get_singleton()->benchmark_dump();
+	}
 
-		if (!defer_load_scene.is_empty()) {
-			OS::get_singleton()->benchmark_begin_measure("Editor", "Load Scene");
+	if (SurfaceUpgradeTool::get_singleton()->is_show_requested()) {
+		SurfaceUpgradeTool::get_singleton()->show_popup();
+	}
 
-			load_scene(defer_load_scene);
-			defer_load_scene = "";
+	// Start preview thread now that it's safe.
+	if (!singleton->cmdline_export_mode) {
+		EditorResourcePreview::get_singleton()->start();
+	}
 
-			OS::get_singleton()->benchmark_end_measure("Editor", "Load Scene");
-			OS::get_singleton()->benchmark_dump();
+	// Load the addons.
+	{
+		_initializing_plugins = true;
+		Vector<String> addons;
+		if (ProjectSettings::get_singleton()->has_setting("editor_plugins/enabled")) {
+			addons = GLOBAL_GET("editor_plugins/enabled");
 		}
 
-		if (SurfaceUpgradeTool::get_singleton()->is_show_requested()) {
-			SurfaceUpgradeTool::get_singleton()->show_popup();
+		for (int i = 0; i < addons.size(); i++) {
+			set_addon_plugin_enabled(addons[i], true);
 		}
+		_initializing_plugins = false;
 
-		// Start preview thread now that it's safe.
-		if (!singleton->cmdline_export_mode) {
-			EditorResourcePreview::get_singleton()->start();
+		if (!pending_addons.is_empty()) {
+			EditorFileSystem::get_singleton()->connect("script_classes_updated", callable_mp(this, &EditorNode::_enable_pending_addons));
 		}
```

Comment:
This part and related code can be removed now (basically reverting #70668), as it's no longer necessary.